js/src/data/api-fetch-middlewares.test.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Internal dependencies
+ */
+import { createErrorResponseCatcher } from './api-fetch-middlewares';
+import { API_NAMESPACE } from './constants';
+
+describe( 'createErrorResponseCatcher', () => {
+	// Ref: https://github.com/WordPress/gutenberg/blob/%40wordpress/api-fetch%405.1.1/packages/api-fetch/src/index.js#L68-L81
+	const checkStatus = ( response ) => {
+		if ( response.status >= 200 && response.status < 300 ) {
+			return response;
+		}
+
+		throw response;
+	};
+
+	const createFetchHandler = ( status, jsonParsedData ) => () => {
+		const response = {
+			status,
+			json() {
+				return Promise.resolve( jsonParsedData );
+			},
+		};
+		return Promise.resolve( response ).then( checkStatus );
+	};
+
+	const passReject = ( response ) => Promise.reject( response );
+
+	const optionsDefaultParse = { path: `${ API_NAMESPACE }/hi` };
+	const optionsDontParse = { path: `${ API_NAMESPACE }/hi`, parse: false };
+
+	describe( 'force overwrite `parse` option to `false` and return the same result', () => {
+		let middleware;
+
+		beforeEach( () => {
+			middleware = createErrorResponseCatcher( passReject );
+		} );
+
+		it( 'should only handle requests where the path belongs to this plugin', async () => {
+			const data = {};
+			const optionsOtherNamespace = { path: '' };
+			const callback = jest.fn().mockImplementation( ( options ) => {
+				expect( options ).toBe( optionsOtherNamespace );
+				return data;
+			} );
```

Comment:
:nail_care: Personally, I like to [name the mocks](https://jestjs.io/docs/mock-functions#mock-names), to make assertion errors more descriptive

Reply:
Updated by https://github.com/woocommerce/google-listings-and-ads/pull/834/commits/4ae6932816b8af405bea9e23869b1d9495c8b3c0

Reply:
Sorry, but that's not what I meant.

By "name the mocks" I mean adding the name property to the mock object, not changing the variable name.

So instead of "anonymous" `mockCallback`
![anonymous mock](https://user-images.githubusercontent.com/17435/124480725-dce1e200-dda7-11eb-86ea-f450673f0632.png)

with `callback = jest.fn()/*...*/.mockName( "middleware's callback" );`
we can have, named `callback`
![image](https://user-images.githubusercontent.com/17435/124481070-31855d00-dda8-11eb-8559-da230a5de5f3.png)

Reply:
Sorry, I didn't read the link's content clearly. Thanks for the elaborate explanation. After I took the https://github.com/woocommerce/google-listings-and-ads/pull/834/commits/f43e6f24d08e5861d9083f44a3dd08db3e1c657a you proposed, the `mockName` was chained to all the mock function calls. 👍